.github/workflows/esp32.yml

```diff
@@ -7,18 +7,17 @@ env:
   CARGO_INCREMENTAL: 0
   RUST_BACKTRACE: short
 
-# by default github use sh as shell
-defaults:
-  run:
-    # will use bash to run each command, bash will source /etc/profile which will give us the environment to build for esp32
-    shell: bash --norc  -leo pipefail {0}
-        
 jobs:
   all:
     name: "Build, clippy and test for esp32"
     runs-on: ubuntu-latest
+    # by default github use sh as shell
```

Comment:
I moved these down a scope, into jobs that have a `container` set. I didn't see much need to override the shell for any other jobs that happened to run outside the container.

Reply:
Fair